vm/devices/user_driver/src/vfio.rs

```diff
@@ -36,7 +36,7 @@ use zerocopy::AsBytes;
 use zerocopy::FromBytes;
 
 pub trait VfioDmaBuffer: 'static + Send + Sync {
```

Comment:
Just sanity checking, there's nothing else that implements this trait today?

Reply:
Today it is LockedMemorySpawner and SharedPoolAllocator. Tomorrow also EmulatedDmaAllocator for unit tests.

Reply:
Do they all follow the new contract?

Reply:
This change benefits normal boot and all clients are compatible. For Restore functionality there is alternative path to restore buffers without zeroiniting them, at least how it's implemented in https://github.com/microsoft/openvmm/pull/210

After we're going to merge with https://github.com/microsoft/openvmm/pull/260 then I'll test end-to-end if it works too.

Reply:
That's not what I asked though. We've added a new requirement to this method, we need to make sure all types that implement this trait are meeting the new requirement.

Reply:
All existing types do (`LockedMemorySpawner` gets it by way of `MAP_ANONYMOUS`). It seems that the test code does not?

https://github.com/microsoft/openvmm/blob/04454925c9fdad1845b4450743b739e16d8a8b4d/vm/devices/user_driver/src/emulated.rs#L168

Reply:
Filed #299